### PR TITLE
SwG release 0.1.22.165

### PR DESCRIPTION
**Pull request by henel677**
## Previous release: [0.1.22.163](https://github.com/subscriptions-project/swg-js/releases/tag/0.1.22.163)

  - Update babel monorepo to v7.14.2 (#1686)
  - Dark button interactive states (#1685)
  - Showcase: Warn about invalid metering state IDs (#1684)
  - Add call to getSubscriptionFlow in GA event listener (#1683)
  - Showcase: Bail if `gaa_at` = 'na' (#1677)
  - Add 'supportsEventManager' to contributions flow iframe (#1682)
  - Update prettier and lint fix (#1681)
  - Update dependency postcss to v8.2.15 (#1680)
  - Update dependency postcss-import to v14.0.2 (#1678)
  - Open a toast "signed in as abc@gmail.com" after verifying entitlement… (#1679)
  - Bump hosted-git-info from 2.8.8 to 2.8.9 (#1675)
  - Update dependency eslint to v7.26.0 (#1674)
  - Showcase: Add additional languages (#1672)
  - Showcase: Styles reset metering demo button. (#1673)
  - Dismiss SwG UI (#1671)
  - Send an empty EntitlementsResponse message to subscription/contributi… (#1669)
  - Compiler: Fixes sourcemaps (#1670)
  - Showcase: Validate timestamps. (#1668)
  - Update dependency mocha to v8.4.0 (#1667)
  - Dialog: Improves test coverage (#1666)
  - Cleanup: Fixes lint issues and formats code across several files (#1665)
  - Clean up some typedefs and update offers documentation (#1664)
  - Fixes lint issues in src/api/subscriptions.js (#1663)
  - Update dependency glob to v7.1.7 (#1662)
  - Update dependency cheerio to v1.0.0-rc.9 (#1660)
  - Update dependency gulp-replace to v1.1.3 (#1661)
  - Close subscription/contribution offers iframe after receiving return … (#1659)
  - Regwall: Add Hindi and Dutch languages (#1658)
  - Update dependency postcss to v8.2.14 (#1657)
  - Add transition states to swg-button (#1656)
  - Force disableNative when paySwgVersion == '2' (#1655)

### CODE REVIEW

**Comment by ChrisAntaki** [comment]
Here's a fixed version of the failing test:
```js
it('should request metering entitlements if URL params are present and timestamp is valid', async () => {
    env.sandbox
      .stub(viewer, 'getReferrerUrl')
      .callsFake(() => Promise.resolve('http://localhost'));
    platform = new GoogleSubscriptionsPlatform(
      ampdoc,
      {enableMetering: true},
      serviceAdapter
    );
    env.sandbox.stub(platform, 'getUrlParams_').returns({
      'gaa_ts': (Date.now() / 1000 + 10).toString(16),
      'gaa_at': 'g',
      'gaa_sig': 'signature',
      'gaa_n': 123456,
    });
    env.sandbox
      .stub(serviceAdapter, 'loadMeteringState')
      .resolves({id: 'abc123'});
    const fetchStub = env.sandbox.stub(xhr, 'fetchJson').callsFake(() =>
      Promise.resolve({
        json: () =>
          Promise.resolve({
            entitlements: [
              {
                source: 'google:metering',
                products: ['example.org:basic'],
                subscriptionToken: 'tok1',
              },
            ],
          }),
      })
    );
    const ents = await platform.getEntitlements();
    expect(ents.service).to.not.be.null;
    expect(ents.source).to.equal('google:metering');

    const fetchUrl = fetchStub.getCall(0).args[0];
    expect(fetchUrl).to.equal(
      'https://news.google.com/swg/_/api/v1/publication/example.org/entitlements?encodedParams=eyJtZXRlcmluZyI6eyJjbGllbnRUeXBlcyI6WzFdLCJvd25lciI6ImV4YW1wbGUub3JnIiwicmVzb3VyY2UiOnsiaGFzaGVkQ2Fub25pY2FsVXJsIjoiMjcwM2YyYjZlZjBlYWFhODEzNzZhMThmYWE3N2E1OTAwOTc1Zjc3MDVkNWQ4YjZlMWEzNzJkNWY2YzJiOTdiYjU5ZjI4M2Q3MzdiNmQ5YWI3N2M1YTNkODQ4YzZlY2UyMDdjZDYwMzU4M2NjMzIyZGQ4MGFiMGI5MzA5MmM2NTAifSwic3RhdGUiOnsiaWQiOiJhYmMxMjMiLCJhdHRyaWJ1dGVzIjpbXX19fQ'
    );
  });
  ```

TLDR: Swgjs no longer sends metering params if they're missing the required metering state ID

**Review by ChrisAntaki — APPROVED** [review]

**Review by dvoytenko — APPROVED** [review]
